### ISSUE TITLE
Add option for displays

### ISSUE THREAD

**Issue by TheRaven500** [OliveTin/OliveTin#73]
It would be nice to have not only buttons but also displays (hope this is the right word?).
Example:
A raspberry pi with a temperatur sensor which writes the temperature to a file (lets say /tmp/kitchen).
In this file you have some text, lets say "21.5°C".
And then maybe something in config like:
```
displays:
- title: Kitchen-Temp
  icon: "&#127777;"
  file: /tmp/kitchen
  poll: 5

```
Poll grabs every X seconds (in this example 5) the value of the file and shows in webbrowser. Or maybe this is not needed because it refreshes it selves?
And of course, there are many other situations which you can use this to show some statuses (light on/off, show water tank level...)
Example (icon 1):
![olivetin](https://user-images.githubusercontent.com/63654062/191801088-45453dcf-273f-4819-8aff-757626e719cb.jpg)

**Comment by jamesread** [comment]
Hey @TheRaven500 , thanks for taking the time to write this up, I can see you've given it some nice thought :-) 

Would the temperature "display" still be a button? In this case, what would happen when you press the button?

Or, would it be better to have a display, and a up/down button?

**Comment by TheRaven500** [comment]
Hi @jamesread , no it is not a button, it is only to show "things" in the web interface :-) I thought "actions" in config could be buttons and "displays" can be only to show things like temperatur or what ever. This way you can mix actions and displays. Maybe you have a better idea to implement such a function.

**Comment by jamesread** [comment]
Hey @TheRaven500 , gotcha - that's what I was thinking as well - "displays", or something like that.

I was working on a prototype of this idea some time ago, called "entities", but never finished it. 

My only worry in implementing this is that I don't think turning OliveTin into a "grafana dashboard with buttons" is the right way to go for the project (I'm a huge fan of Grafana), but I appreciate what you're trying to do isn't exactly the Grafana use case.

On reflection, having a command executing in the background, updating the text of the button, could allow some really interesting use cases for builders, makers, IoT devices and similar - allowing people to easily build quick user interfaces. Hmm. This could be exciting, but needs a little more time for thought on how that design would work. I'll come back to this when I'm able. 

If other people are reading this, thinking it's interesting, please :+1: or :-1: on the first comment, as a way of tracking interest in this. Thanks @TheRaven500 .

**Comment by TheRaven500** [comment]
Hi @jamesread i also played with the idea to change the icon when a button is pressed (with the play and stop symbol which starts or stops timelapse on my camera). I tried to do this (yes i know it is VERY ugly!) with my script which changes the config file of OliveTin and replaces the symbol with the "sed" command. It has not worked because sed does not "save" the file. Because of that OliveTin does not recognize it and does not update the button. There is an option in sed to "save" the file, but i haven't found the right command so far. It looks like sed is too complex for me.
And yes you are absolutely right, OliveTin should be simple and not as complex like grafana.
Sometimes its good to "brainstorm" about something, and maybe some nice thing is born. ;-)

**Comment by jamesread** [comment]
Hey @TheRaven500 , OK, I can see that being useful too actually. Perhaps a REST API call to change button text/icon would be useful here. I've actually created a new issue for that; https://github.com/OliveTin/OliveTin/issues/74 .

Your suggested workaround should work - you should be able to use sed "edit the [config] file in-place", I think it's something like `sed -i "s/old title/new title" /etc/OliveTin/config.yaml` - but you're right, this is **very ugly**!

**Comment by TheRaven500** [comment]
Ok, i have it working but it is not so easy. The sed command does not "update" the file, because of that OliveTin does not recognize the change and does not update the button. But i found a work a round (as always...).
Config:
- title: Temp 23.0 # Title1
shell: sleep 0.5
icon: "&#127777;"

Script to update:
#!/bin/bash
sed '/Title1/c\- title: Temp 23.0 # Title1' /etc/OliveTin/config.yaml > /tmp/config.yaml && cp /tmp/config.yaml /etc/OliveTin/config.yaml

Title1 is a placeholder to identify which action should be updated. This way you can have multiple actions to change. With the copy command, OliveTin recognizes the file change and update the action button immediately.
Maybe it helps someone who had the same problem.

**Comment by jamesread** [comment]
Hey @TheRaven500 , heh, your hack works, but my suggestion in my last comment was just to use `-i` to "edit the file in place" - that way you can avoid the temporary file solution. 

However, I'm going to implement a proper API for doing this instead - so you can avoid hacks in the future. :-)

**Comment by TheRaven500** [comment]
In file edit does NOT work, OliveTin does not recognize the change in the file, because of that it does NOT update the button in the webgui. I have also tried to move the file with the mv command, but only cp works. But yes, it would be nice not to do such "Q & D" things in the future! :-)

**Comment by jamesread** [comment]
So, I finally wrote the code for this and released it in version 2024.02.27. Lots of this design was inspired by this issue and your use case, so I hope this is what you wanted!

![image](https://github.com/OliveTin/OliveTin/assets/2025715/6b8bbbc3-4e4d-4763-9aa0-914a7ecc3ed4)

I documented how to do this here on this page; 

https://docs.olivetin.app/heating-control-panel.html

I'm sorry that it took just over a year! Can you let me know if this works OK for you? If so, I will finally close this issue!
